library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
Couldn't we get into and endless loop here?
What will happen if the peer socket gets disconnected, so f_recv will always return failure, causing:
~~~
     if( ( ret = mbedtls_ssl_fetch_input( ssl, mbedtls_ssl_hdr_len( ssl ) ) ) != 0 )
      {		      {
 @@ -3815,8 +3818,7 @@ int mbedtls_ssl_read_record_layer( mbedtls_ssl_context *ssl )
                                              "(header)" ) );		                                              "(header)" ) );
              }		              }
  		  
 -            /* Get next record */		 +            return( MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
 -            goto read_record_header;		
          }		          }
~~~
inside `mbedtls_ssl_read_record_layer`.